### PR TITLE
Client-side "Stop" API method for DataFrame analytics config.

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR implements "Stop" method for DataFrame analytics configs in high-level REST client.

### CODE REVIEW

File: client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningIT.java [hunk 1]
```diff
@@ -1388,6 +1390,45 @@ public void testStartDataFrameAnalyticsConfig() throws Exception {
...
+        String sourceIndex = "stop-test-source-index";
+        String destIndex = "stop-test-dest-index";
+        createIndex(sourceIndex);
+        highLevelClient().index(new IndexRequest(sourceIndex).source(XContentType.JSON, "total", 10000), RequestOptions.DEFAULT);
+
+        // Verify that the destination index does not exist. Otherwise, analytics' reindexing step would fail.
+        assertFalse(highLevelClient().indices().exists(new GetIndexRequest(destIndex), RequestOptions.DEFAULT));
+
+        MachineLearningClient machineLearningClient = highLevelClient().machineLearning();
+        String configId = "start-test-config";
```
Comment (benwtrent) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningIT.java]:
```suggestion
        String configId = "stop-test-config";
```
nit

**Review by benwtrent — COMMENTED** [review]
Please add the HLRC docs as well.

**Comment by przemekwitek** [comment]
> Please add the HLRC docs as well.

@benwtrent: I added "Evaluate API" docs to the docs PR: https://github.com/elastic/elasticsearch/pull/42288

**Comment by przemekwitek** [comment]
> Please add the HLRC docs as well.

@benwtrent: I'm planning to add docs to my docs PR (https://github.com/elastic/elasticsearch/pull/42288)

**Review by benwtrent — APPROVED** [review]

**Comment by przemekwitek** [comment]
run elasticsearch-ci/1

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/StopDataFrameAnalyticsAction.java [hunk 2]
```diff
@@ -132,13 +132,11 @@ public void writeTo(StreamOutput out) throws IOException {
...
         @Override
         public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-            builder.startObject();
-            if (id != null) {
-                builder.field(DataFrameAnalyticsConfig.ID.getPreferredName(), id);
-            }
-            builder.field(ALLOW_NO_MATCH.getPreferredName(), allowNoMatch);
-            builder.endObject();
-            return builder;
+            return builder
```
Comment (dimitris-athanasiou) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/StopDataFrameAnalyticsAction.java]: Just a note that I don't think should change this but it's useful for the future.

I think the reason `toXContent` is implemented around the codebase mostly without the chained paradigm but with separating each step into its own statement is to make it easy to add statements that need to be surrounded with null checks. While the chained paradigm is nice and compact, if we need to add in the future a null-check for a certain field, we'll need to refactor the whole method.

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/StopDataFrameAnalyticsAction.java (hunk 2 above)
Comment (przemekwitek) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/StopDataFrameAnalyticsAction.java]: Ack

**Comment by jpountz** [comment]
@przemekwitek FYI All merged PRs should have one of the `>feature`, `>enhancement`, ... labels for release notes generation to work properly. Also, no need to add this information to the PR title.

**Comment by przemekwitek** [comment]
> @przemekwitek FYI All merged PRs should have one of the `>feature`, `>enhancement`, ... labels for release notes generation to work properly. Also, no need to add this information to the PR title.

Ack, thanks.

**Pull request merged**